.github/workflows/test.yaml

```diff
@@ -191,9 +214,35 @@ jobs:
         git reset --hard $BRANCH_SHA
 
     - name: Newsfragment
+      if: ${{ !cancelled() }}
       run: |
         tox -q -e newsfragment
 
+    # This is called only on code changes.
+    # If you forgot to define the right PR title you should re-run the job.
+    - name: Check PR title format
+      if: github.event_name == 'pull_request'
+      shell: python
+      env:
+        PR_TITLE: ${{ github.event.pull_request.title }}
+      run: |
+        import os, re, sys
```

Comment:
IMO code in a yaml file is an anti-pattern.  At most this should reference an external tool that can be a regular Python file (and so gain editor support for proper 4 space indents :wink: ).

Reply:
The thing is that this code is not very useful outside of GHA.

But you are right. I will extract it to make it easier to test it